src/DynamoCoreWpf/Utilities/NodeContextMenuBuilder.cs

```diff
@@ -0,0 +1,451 @@
+            //textBlock.SetValue(VerticalAlignmentProperty, VerticalAlignment.Center);
+            //textBlock.SetValue(ForegroundProperty, new SolidColorBrush(Color.FromRgb(238,238,238)));
+
+            //dismissedAlertsMenuItem.ItemTemplate = itemTemplate;
+
+            //AddContextMenuItem(dismissedAlertsMenuItem, insertionPoint++);
+
+            return null;
+        }
+
+        private static void AddInjectedNodeViewCustomizationMenuItems(Dictionary<string, object> nodeViewCustomizationMenuItems)
+        {
+            foreach (KeyValuePair<string, object> keyValuePair in nodeViewCustomizationMenuItems)
+            {
+                ContextMenu.Items.Add(keyValuePair.Value);
+            }
+        }
+
+        private static MenuItem BuildIsInputMenuItem()
+        {
+            return CreateMenuItem
+            (
+                name: "isInput_cm",
+                header: Properties.Resources.NodeContextMenuIsInput,
+                isCheckable: true,
+                isChecked: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("IsSetAsInput"),
+                    Mode = BindingMode.TwoWay,
+                }
+            );
+        }
+
+        private static MenuItem BuildIsOutputMenuItem()
+        {
+            return CreateMenuItem
+            (
+                name: "isOutput_cm",
+                header: Properties.Resources.NodeContextMenuIsOutput,
+                isCheckable: true,
+                isChecked: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("IsSetAsOutput"),
+                    Mode = BindingMode.TwoWay,
+                }
+            );
+        }
+
+        private static MenuItem BuildHelpMenuItem()
+        {
+            return CreateMenuItem
+            (
+                name: "help_cm",
+                header: Properties.Resources.NodeContextMenuHelp,
+                command: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("ShowHelpCommand")
+                }
+            );
+        }
+
+        /// <summary>
+        /// A list of the names of nodes' default menu items, which cannot be dynamically injected into its context menu.
+        /// </summary>
+        internal static readonly List<string> NodeContextMenuDefaultItemNames = new List<string>
+        {
+            Properties.Resources.ContextMenuDelete,
+            Properties.Resources.ContextMenuGroups,
+            Properties.Resources.NodeContextMenuPreview,
+            Properties.Resources.NodesRunStatus,
+            Properties.Resources.NodeContextMenuShowLabels,
+            Properties.Resources.NodeContextMenuRenameNode,
+            Properties.Resources.ContextMenuLacing,
+            //Wpf.Properties.Resources.NodeInformationalStateDismissedAlerts,
```

Comment:
??

Reply:
This is intentionally commented out, to be tied together with the node info states once that work has been merged.